docs/core/diagnostics/cli-tools/dotnet-trace.md

```diff
@@ -0,0 +1,182 @@
+  - On Linux, the trivial option could be using `pidof` on the terminal window.
+
+You may also use the command `dotnet-trace list-processes` command to find out what .NET Core processes are running, along with their process IDs.
+
+- Then, run the following command:
+
+```bash
+dotnet-trace collect --process-id <PID> --providers Microsoft-Windows-DotNETRuntime
+
+Press <Enter> to exit...
+Connecting to process: <Full-Path-To-Process-Being-Profiled>/dotnet.exe
+Collecting to file: <Full-Path-To-Trace>/trace.nettrace
+  Session Id: <SessionId>
+  Recording trace 721.025 (KB)
+```
+
+- Finally, stop collection by pressing the \<Enter> key, and `dotnet-trace` will finish logging events to *trace.nettrace* file.
+
+### Using `dotnet-trace` to collect counter values over time
+
+If you're trying to use `EventCounter` for basic health monitoring in  performance-sensitive settings like production environments and you want to collect traces instead of watching them in real time, you can do that with `dotnet-trace` as well.
+
+For example, if you want to collect runtime performance counter values, you can use the following command:
+
+```bash
+dotnet-trace collect --process-id <PID> --providers System.Runtime:0:1:EventCounterIntervalSec=1
+```
+
+This command will tell the runtime counters to be reported once every second for lightweight health monitoring. Replacing `EventCounterIntervalSec=1` with a higher value (say 60) will allow you to collect a smaller trace with less granularity in the counter data.
+
+If you want to disable runtime events to reduce the overhead (and trace size) even further, you can use the following command to disable runtime events and managed stack profiler.
+
+```bash
+dotnet-trace collect --process-id <PID> --providers System.Runtime:0:1:EventCounterIntervalSec=1,Microsoft-Windows-DotNETRuntime:0:1,Microsoft-DotNETCore-SampleProfiler:0:1
+```
+
+## Viewing the trace captured from `dotnet-trace`
+
+On Windows, `.nettrace` files can be viewed on [PerfView](https://github.com/microsoft/perfview) for analysis, just like traces collected with ETW or LTTng. For traces collected on Linux, you can either move the trace to a Windows machine to be viewed on PerfView.
+
+You may also view the trace on a Linux machine by changing the output format of `dotnet-trace` to `speedscope`. You can change the output file format using the `-f|--format` option - `-f speedscope` will make `dotnet-trace` to produce a `speedscope` file. You can currently choose between `nettrace` (the default option) and `speedscope`. `Speedscope` files can be opened at https://www.speedscope.app.
+
+Note: The .NET Core runtime generates traces in the `nettrace` format, and they're converted to speedscope (if specified) after the trace is completed. Since some conversions may result in loss of data, the original `nettrace` file is preserved next to the converted file.
+
+## Commonly used keywords for the *Microsoft-Windows-DotNETRuntime* provider
```

Comment:
Move this below the list of .NET providers below? There is also probably better documentation on these events here: https://docs.microsoft.com/en-us/dotnet/framework/performance/clr-etw-keywords-and-levels so if we can link to that info rather than replicating a summary that would probably be easier to maintain in the long run. We may need to clarify that even though that page refers to the events as "ETW" events they are also valid in this context.